builder/cloudstack/step_create_instance.go

```diff
@@ -65,7 +115,29 @@ func (s *stepCreateInstance) Run(state multistep.StateBag) multistep.StepAction
 	}
 
 	if config.UserData != "" {
-		ud, err := getUserData(config.UserData, config.HTTPGetOnly)
+		var httpPort uint = state.Get("http_port").(uint)
+		hostIp, err := hostIP()
+		if err != nil {
+			state.Put("error", err)
+			ui.Error(err.Error())
+			return multistep.ActionHalt
+		}
+		common.SetHTTPIP(hostIp)
```

Comment:
This is to fill in the variables ```HTTPIP``` and ```HTTPPort```in the template rendering for user_data, so we can say ```ks=http://{{ .HTTPIP }}:{{ .HTTPPort }}/kickstart.cfg```. Is it okay, @svanharmelen ?